### PR TITLE
Ngram (Week 3 Day 4) and Citibike (Week 3 Day 5 & Week 4 Day 1)

### PR DESCRIPTION
**Pull request by jcqln0**
jackies pile of work (ngrams and citibike prediction)
- forgot to PR ngram stuff
- all of the citibike

### CODE REVIEW

**Comment by hannahcha417** [comment]
Nice work on the ngrams assignment!! I'll be back to this PR to grade the citibike work

File: week3/predict_citibike.Rmd [hunk 1]
```diff
@@ -40,29 +89,352 @@ Add features that might help: day of week, month, precipitation, etc.
...
+# Possible model features to include
+# Current features: ymd, num_trips, date, precipitation, snow depth, snow, max temp, min temp
+# Possibly interesting features that contribute to trips
+# tmin (colder temps)
+# tmax (hotter temps)
+# day of the week (weekends vs weekdays) and month of the year
+trips_per_day_kfold <- mutate(trips_per_day_kfold, day_of_week = wday(ymd), month = month(ymd))
+# snow depth (substantial snow depth i decided over 5) and precipitation (substantial precipitation decided for over 0.5)
+trips_per_day_kfold <- mutate(trips_per_day_kfold, snow_sub = (snwd > 5), prcp_sub = (prcp > 0.5))
+
```
Comment (hannahcha417) [inline week3/predict_citibike.Rmd:101]: Nice features!

File: week3/predict_citibike.Rmd [hunk 2]
```diff
@@ -73,4 +445,11 @@ Pick your best model and plot the results.
...
 # TODO: compute RMSE on test set
 # TODO: how do you think this model will perform on future data?
+trips_per_day_test <- mutate(trips_per_day_kfold, day_of_week = wday(ymd), month = month(ymd))
+
+load('final_model.RData')
+rmse(final_model, trips_per_day_test)
+rsquare(final_model, trips_per_day_test)
+# My RMSE for the test set was 3208.211
+# I think the model will perform okay? I'm a little concerned about if I overfit the training/validation data
 ```
```
Comment (hannahcha417) [inline week3/predict_citibike.Rmd:455]: Generally, your performance on the validation set should be somewhat close to your test set performance!

**Comment by hannahcha417** [comment]
Great job on the citibike assignment! You were very thoughtful in designing your models; nice work!
